SPIE2016-9913-16/9913-16.tex

```diff
@@ -0,0 +1,303 @@
+Both the Astropy and IVOA communities would benefit if such a standard could be established and it could be a significant factor in bringing together the competing distortion models currently in use in the FITS community.
+
+\subsection{Region handling}
+
+The handling of regions, such as circles and polygons, in either Cartesian pixel coordinates or spherical world coordinates or with discrete pixelized or analytic form (Table~\ref{tab:regions}), is not yet supported in Astropy core.
+The \texttt{photutils} affiliated Astropy package has some basic support for analytic regions in pixel coordinates for photometry and work has recently started on a regions package that can properly handle spherical geometry and world coordinate versions translated to pixels.
+LSST has its own spherical geometry package in C++, \texttt{lsst::sphgeom}, and we should compare implementations with the STScI spherical geometry library (mainly written in Python) that is being proposed for integration into Astropy.
+There is currently a focus within Astropy on handling region files from the DS9 application\cite{2005ASPC..347..110J} but there have also been discussions concerning adopting the IVOA STC standard\cite{2007ivoa.spec.1030R} as a region serialization format.
+STC-S is already being used to define sky regions in the VO community and is also understood by the GAIA visualization tool\cite{2009ASPC..411..575D} and Starlink software\cite{2010ASPC..434..213B,2014ASPC..485..391C}.
+If Astropy and LSST can not share sky to pixel region handling code directly there is a possibility that STC could be adopted to allow interoperability.
+
+\begin{table}[t]
+\centering
+\caption{Four distinct types of region. This paper discusses all except discrete spherical.}
+\label{tab:regions}
+\vspace{10pt}
+\begin{tabular}{ccc}
+  Discrete Spherical & $\Longleftrightarrow$ & Discrete Cartesian \\
+  $\Updownarrow$ & &  $\Updownarrow$ \\
+  Geometric Spherical & $\Longleftrightarrow$ & Geometric Cartesian \\
+\end{tabular}
+\end{table}
+
+Astropy has no plans for a discretized efficient representation of a pixel region and this is something that LSST can contribute to the Astropy region handling package.
+In LSST a \emph{footprint} is a compact serialized representation of a mask, listing the pixel indices that contribute to the shape in rasterized order, using a run-length encoding algorithm to conserve space.
+A \emph{heavy footprint} is a \emph{footprint} that includes the pixel values themselves.
+These are the items that can be passed to measurement algorithms.
+\emph{Footprints} are used extensively in the LSST code base and would be a useful addition to Astropy.
+
+There are no plans currently in either project to handle sky pixelization representations of regions, such as multi-order coverage maps (MOC)\cite{2012ASPC..461..347F,2014ivoa.spec.0602F} or other approaches\cite{2015A&A...580A.132R}.
```

Comment:
Developing or adopting a library for manipulating pixelized spherical regions in Science Pipelines and Level 3 Toolkit code is also very much in the LSST construction plan; we've just been planning to do that relatively late in construction in the hopes this improves our chances of being able to use a third-party library to do more (or, at least, that's the excuse on the Science Pipelines side).  So it might actually be worth calling this out as something LSST ultimately needs to do.